docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -2053,303 +2053,285 @@ The following error message codes may appear on the app-server log. Use the foll
+  Plugin (_PluginIdentifier_) loading failed. Message: "_pluginsTotal_" Successful: _pluginsLoaded_% (_pluginsLoaded_/_eventCount_) Attempted: _pluginCount_% (_pluginCount_/_eventCount_)
 
   **Reason:**
 
-  TODO
+  Plugin with _pluginId_ loaded failed. The server attempted to load a total of _pluginsTotal_ with _pluginsLoaded_ plugins already successfully loaded.
 
   **Action:**
 
-  TODO
+  Review _errorMessage_ to see if there is something you can do to fix the error. You may need to contact the plugin developer to find a solution. If you do not need this plugin, it is OK to continue.
 
 
 
-### ZWED0165W":"Unable to retrieve storage data from cluster promise %s
+### ZWED0166W
+
+  Error updating the storage: _Error_
 
   **Reason:**
 
-  TODO
+  Throws warning _Error_ when it faced error while updating the storage.
 
   **Action:**
 
-  TODO
+  Contact support if _Error_ is not clear.
+
 
 
+### ZWED0167W
 
-### ZWED0166W":"Error updating the storage: %s
+  Error adding to the storage: _storageTimeout_
 
   **Reason:**
 
-  TODO
+  Throws _storageTimeout_ error while adding to the storage.
 
   **Action:**
 
-  TODO
+  You may change the value of _storageTimeout_ ms by setting 'node.cluster.storageTimeout' within the config.
+
 
 
+### ZWED0168W
 
-### ZWED0167W":"Error adding to the storage: %s
+  Unable to retrieve storage value from cluster _Error_
 
   **Reason:**
 
-  TODO
+  Throws warning _Error_ when it unable to retrieve storage value from cluster.
 
   **Action:**
 
-  TODO
+  Contact support if _Error_ is not clear.
+
 
 
+### ZWED0169W
 
-### ZWED0168W":"Unable to retrieve storage value from cluster %s
+  Error deleting the storage with id: _deleteStorageByKey_  _Error_
 
   **Reason:**
 
-  TODO
+  Throws warning _Error_ when it tries deleting the storage with _deleteStorageByKey_ 
 
   **Action:**
 
-  TODO
+  Contact support if _Error_ is not clear.
+
 
 
+### ZWED0170W
 
-### ZWED0169W":"Error deleting the storage with id: %s %s
+  Plugin (_PluginIdentifier_) loading failed. Version: _PluginVersion_. Message: "_Error_"
 
   **Reason:**
 
-  TODO
+  Throws warning _Error_ when plugin with _PluginIdentifier_ failed to load. version of plugin is _PluginVersion_. 
 
   **Action:**
 
-  TODO
+  Review _Error_ to see if there is something you can do to fix the error. You may need to contact the plugin developer to find a solution. If you do not need this plugin, it is OK to continue.
 
 
 
-### ZWED0170W":"Plugin (%s) loading failed. Version: %s. Message: "%s"
 
-  **Reason:**
+### ZWED0171W
 
-  TODO
-
-  **Action:**
-
-  TODO
-
-
-
-### ZWED0171W":"Rejected undefined referrer for url=%s, ip=%s
+  Rejected undefined referrer for url=_originalUrl_, ip=_ip_
 
   **Reason:**
 
-  TODO
+  Throws 403 Forbidden due to referrer, when it failed to load the webApp. 
 
   **Action:**
 
-  TODO
+  Double Check the Address. The most common reason for a 403 error is a mistyped _originalUrl_ or _ip_.
```

Comment:
*Double check the address. A possible reason for a 403 error is a misstyped _originalUrl_ or _ip_ or because loopback routing is not configured in the App server.